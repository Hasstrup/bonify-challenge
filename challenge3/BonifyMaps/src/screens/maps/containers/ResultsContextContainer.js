import React from "react";
import PropTypes from "prop-types";
import ResultsContextView from "../components/ResultsContext/";
import {
  VIDEO_RESULTS_FETCHED,
  NEW_USER_CONTEXT,
  UNRESOLVED_COORDINATES_CONTEXT
} from "src/Constants";

/**
 *
 * @export
 * @name ResultsContextContainer
 * @desc container housing all the methods necessary for interacting
 *  with the results from the user's interaction with google maps.
 *
 */
const ResultsContextContainer = props => {
  /**
   * @name handleNavigation
   * @desc handles routing from the maps index page
   * to the videos index page passing the latLng, so the
   * next screen can find the
   * @returns {null}
   */
  const handleNavigation = () => {
    props.navigator.push({
      component: {
        name: "VideosIndexScreen",
        passProps: {
          location: props.location
        }
      }
    });
  };

  return (
    <ResultsContextView
      currentUIContext={VIDEO_RESULTS_FETCHED}
      address={props.address}
      processing={props.processing}
      handleButtonPress={handleNavigation}
    />
  );
};

ResultsContextContainer.propTypes = {
  address: PropTypes.string
};
export default ResultsContextContainer;
