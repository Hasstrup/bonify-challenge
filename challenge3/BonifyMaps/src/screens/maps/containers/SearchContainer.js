import React, { useState } from "react";
import PropTypes from 'prop-types';
import SearchContextView from "../components/SearchContext";
import * as MapActions from "src/Actions";

/**
 * 
 * @name SearchContiner
 * @desc handles the underlying logic for the search feature
 * @param {any} props 
 * @returns 
 */
const SearchContainer = props => {
  const [searchText, setSearchText] = useState("");
  const onInputChange = text => setSearchText(text);
  const onButtonPress = () => {
    const { handleAddressChange, handleErrorCallback } = props;
    MapActions.handleAddressChange(
      searchText,
      handleAddressChange,
      handleErrorCallback
    );
  };
  return (
    <SearchContextView
      onTextInputChange={onInputChange}
      searchText={searchText}
      onButtonPress={onButtonPress}
    />
  );
};

SearchContainer.PropTypes = {
    handleAddressChange: PropTypes.func,
    handleErrorCallback: PropTypes.func
}
export default SearchContainer;
