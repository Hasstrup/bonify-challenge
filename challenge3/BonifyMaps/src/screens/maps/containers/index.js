import MapContainer from './mapContainer';

export {
    MapContainer
}