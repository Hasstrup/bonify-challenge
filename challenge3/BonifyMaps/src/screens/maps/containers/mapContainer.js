import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import Map from "../components/Map/Map";
import * as MapActions from "src/Actions";

/**
 *
 * @name MapContainer
 * @desc handles the data and manipulation logic
 * for map interactions
 * @param {any} props
 * @returns
 */
const MapContainer = props => {
  const [usersLocation, setUsersLocation] = useState({});
  useEffect(() => {
    MapActions.getUsersLocation(setUsersLocation);
  }, []);

  /**
   *
   * @name onLocationEventChange
   * @desc listener for change events on the map, including when
   * a user presses the map or stops dragging. Makes a call to the service
   * after the trigger
   * @param {any} coords coords containing latitude and longitude
   */
  onLocationEventChange = coords => {
    const { handleAddressChange, handleErrorCallback } = props;
    MapActions.handleLocationChange(
      coords,
      handleAddressChange,
      handleErrorCallback
    );
  };

  return (
    <Map
      usersRegion={usersLocation}
      handleLocationChange={onLocationEventChange}
    />
  );
};

MapContainer.propTypes = {
  handleAddressChange: PropTypes.func.isRequired,
  handleErrorCallback: PropTypes.func.isRequired
};

MapContainer.defaultProps = {
  handleAddressChange: r => {
    console.log(r);
  },
  handleErrorCallback: () => {}
};

export default MapContainer;
