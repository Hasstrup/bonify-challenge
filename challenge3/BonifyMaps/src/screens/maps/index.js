import React, { useState, useEffect } from "react";
import { View } from "react-native";
import PropTypes from 'prop-types';
import { setNavigatorProp } from "src/navigation/actions";
import * as MapActions from "src/Actions";
import {
  MapContainer,
  ResultsContextContainer,
  SearchContainer
} from "src/screens/maps/containers";

/**
 *
 * @name MapsIndexScreen
 * @desc Component rendered for on the index screen.
 * @param {any} props
 * @returns {func} React Component
 */
export const MapsIndexScreen = props => {
  const navigator = setNavigatorProp(props.componentId);

  const initialState = {
    region: {
      latitude: 37.78825,
      longitude: -122.4324,
      latitudeDelta: 0.0922,
      longitudeDelta: 0.0421
    },
    address: "Fetching your location ...",
    latLng: "",
    processing: false,
    errors: false
  };

  const [state, setState] = useState(initialState);

  useEffect(() => {
    MapActions.getUsersLocation(region => {
      setState({ ...state, region });
    });
  }, []);

  const handleAddressChange = ({ region, latLng, address }) => {
    setState({
      ...state,
      region,
      latLng,
      address,
      processing: false,
      errors: false
    });
  };

  const handleMapInteraction = region => {
    setState({ ...state, processing: true });
  };

  return (
    <View style={{ flex: 1, position: "relative" }}>
      <MapContainer
        region={state.region}
        indicateProcess={() => setState({ ...state, processing: true })}
        handleAddressChange={handleAddressChange}
        handleMapInteraction={handleMapInteraction}
        handleErrorCallback={err =>
          setState({
            ...state,
            errors: true,
            address: err.message,
            processing: false
          })
        }
      />
      <ResultsContextContainer
        processing={state.processing}
        errors={state.errors}
        navigator={navigator}
        address={state.address}
        location={state.latLng}
      />
      <SearchContainer
        handleAddressChange={handleAddressChange}
        handleErrorCallback={err =>
          setState({ ...state, errors: true, address: err.message })
        }
      />
    </View>
  );
};

/**
 *
 *
 * @static
 * @param {any} passProps
 * @returns {object} the navigation options for the class
 * @desc set of navigation options that determine the layout of the code
 * @memberOf MapsIndexScreen
 */
MapsIndexScreen.options = () => ({
  topBar: {
    visible: false,
    drawBehind: true
  }
});

MapsIndexScreen.propTypes = {
  componentId: PropTypes.string
}
