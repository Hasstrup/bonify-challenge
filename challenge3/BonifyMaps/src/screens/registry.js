/** 
 * The order of imports matter, since we are using a stack navigator.
 *  The first screen imported will be the root screen
*/
export { MapsIndexScreen } from './maps';
export { VideosIndexScreen } from './videos/index'
export { VideosShowScreen } from './videos/show'
