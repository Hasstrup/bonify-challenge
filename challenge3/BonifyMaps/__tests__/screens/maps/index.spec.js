import 'react-native';
import React from 'react';
import { shallow } from 'enzyme'
import { MapsIndexScreen } from 'src/screens/registry'
import { MapContainer } from 'src/screens/maps/containers/'


describe('Maps Index screen', () => {
    it('renders correctly and matches snapshot', () => {
        const wrapper = shallow(<MapsIndexScreen />)
        expect(wrapper).toMatchSnapshot()
        wrapper.setProps({ componentId: "testComponentId" })
        expect(wrapper).toMatchSnapshot()
    })

    it('contains the MapContainer', () => {
        const wrapper = shallow(<MapsIndexScreen />);
        expect(wrapper.find(MapContainer)).toExist()
    })

})