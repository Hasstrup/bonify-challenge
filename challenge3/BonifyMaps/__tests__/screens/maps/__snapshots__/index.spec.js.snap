// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps Index screen renders correctly and matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <MapContainer
    handleAddressChange={[Function]}
    handleErrorCallback={[Function]}
    handleMapInteraction={[Function]}
    indicateProcess={[Function]}
    region={
      Object {
        "latitude": 37.78825,
        "latitudeDelta": 0.0922,
        "longitude": -122.4324,
        "longitudeDelta": 0.0421,
      }
    }
  />
  <ResultsContextContainer
    address="Fetching your location ..."
    errors={false}
    location=""
    navigator={
      Object {
        "pop": [Function],
        "push": [Function],
      }
    }
    processing={false}
  />
  <SearchContainer
    handleAddressChange={[Function]}
    handleErrorCallback={[Function]}
  />
</View>
`;

exports[`Maps Index screen renders correctly and matches snapshot 2`] = `
<View
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <MapContainer
    handleAddressChange={[Function]}
    handleErrorCallback={[Function]}
    handleMapInteraction={[Function]}
    indicateProcess={[Function]}
    region={
      Object {
        "latitude": 37.78825,
        "latitudeDelta": 0.0922,
        "longitude": -122.4324,
        "longitudeDelta": 0.0421,
      }
    }
  />
  <ResultsContextContainer
    address="Fetching your location ..."
    errors={false}
    location=""
    navigator={
      Object {
        "pop": [Function],
        "push": [Function],
      }
    }
    processing={false}
  />
  <SearchContainer
    handleAddressChange={[Function]}
    handleErrorCallback={[Function]}
  />
</View>
`;
